CMakeLists.txt

```diff
@@ -71,6 +71,15 @@ set(ZEPHYR_CURRENT_LINKER_PASS 0)
 set(ZEPHYR_CURRENT_LINKER_CMD    linker_zephyr_pre${ZEPHYR_CURRENT_LINKER_PASS}.cmd)
 set(ZEPHYR_LINK_STAGE_EXECUTABLE zephyr_pre${ZEPHYR_CURRENT_LINKER_PASS})
 
+# Make kconfig variables available to the linker script generator
+zephyr_linker_include_generated(FILE ${CMAKE_CURRENT_BINARY_DIR}/.config KCONFIG)
+
+# The linker generator also needs sections.h to be able to access e.g. _APP_SMEM_SECTION_NAME
+# for linkerscripts that do not support c-preprocessing.
+zephyr_linker_include_generated(FILE ${ZEPHYR_BASE}/include/zephyr/linker/sections.h)
```

Comment:
not to be changed in this PR.

I consider that perhaps it's better to have developers being able to describe the content of `sections.h` in CMake or a template file which CMake can use, and from there generate header files, CMake files, and whatever else is needed by the build system.

CMake is better at generating header files than reading them :wink: 

Thoughts ?